src/State/WorldMap/WorldMap.h

```diff
@@ -0,0 +1,157 @@
+                    signed int tileDimension,
+                    signed int mapDimension
+                );
+
+                bool inRenderView(
+                    signed int worldTileMin,
+                    signed int delta,
+                    signed int tileDimension,
+                    signed int mapDimension
+                );
+
+                void travel();
+                int  setDestination(char orientation, unsigned int clickPosition);
+                void preventTravelOutboundMin(float* nextPosition);
+                void preventTravelOutboundMax(float* nextPosition, float currentPosition, int maxRange);
+
+            private:
+                //get keyboard state pointer
+                const Uint8* keystate = SDL_GetKeyboardState(NULL);
+                UI::ImageList* _tiles = nullptr;
+                UI::ImageButton* _hotspot = nullptr;
+
+                //panel
+                UI::Image* _panel = nullptr;
+                Graphics::Size panelSize;
+                unsigned int panelX;
+                unsigned int panelY;
+                unsigned int fullscreenSidepanelWidth;
+                unsigned int panelBorder = 22;
+
+                // temporary!
+                // @todo: move it to other place!
+                // coordinates of the player on world map, has car
+                float worldMapX = 0;
+                float worldMapY = 0;
+                float nextWorldMapX = 0;
+                float nextWorldMapY = 0;
+                bool hasCar = true;
+                float directionHorizontal = 0;
+                float directionVertical = 0;
+
+                unsigned int destinationX = 0;
+                unsigned int destinationY = 0;
+                bool isTraveling = false;
+
+                // map offset for fullscreen or windowed
+                signed int offsetX = 0;
+                signed int offsetY = 0;
+
+                // delta (shift of map)
+                signed int deltaX = 0;
+                signed int deltaY = 0;
+
+                // fallout2 map size: 4x5 tiles, so texture size is 1400x1500
+                unsigned int tilesNumberX = 4; // fallout 2 tiles by X
+                unsigned int tilesNumberY = 5; // fallout 2 tiles by Y
+                unsigned int tileWidth = 350;  // fallout 2 map tile width
+                unsigned int tileHeight = 300; // fallout 2 map tile height
+
+                // default size data
+                unsigned int mapWidth = 450;   // fallout 2 map screen width
+                unsigned int mapHeight = 442;  // fallout 2 map screen height
+                unsigned int mapMinX;      // start X point of map screen
+                unsigned int mapMinY;      // start Y point of map screen
+
+                const std::string sidePanelSprite = "art/intrface/wminfce2.frm";
+                const std::string fullPanelSprite = "art/intrface/wmapbox.frm";
+
+                std::vector<std::string> _worldMapTiles = {
```

Comment:
Perhaps you could use `std::array` here instead? Someone suggested that in one of my previous commits, so I am passing it on :smile:

Reply:
You don't need any arrays here, it should be read from the intrface.lst file
_Do you specifically write a non-compatible engine?_